olive/passes/onnx/genai_model_exporter.py

```diff
@@ -0,0 +1,76 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# Export a PyTorch model using the onnxruntime-genai package.
+# --------------------------------------------------------------------------
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict
+
+from onnx import TensorProto
+
+from olive.hardware.accelerator import AcceleratorLookup, AcceleratorSpec, Device
+from olive.model import ONNXModelHandler, PyTorchModelHandler
+from olive.model.utils import resolve_onnx_path
+from olive.passes import Pass
+from olive.passes.olive_pass import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class GenAIModelExporter(Pass):
+    class Precision(str, Enum):
+        FP32 = "fp32"
+        FP16 = "fp16"
+        INT4 = "int4"
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "precision": PassConfigParam(
+                type_=GenAIModelExporter.Precision,
+                required=True,
+                description="Precision of model.",
+            )
+        }
+
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from onnxruntime_genai.models.builder import create_model
```

Comment:
I tried to go through the onnxruntime-genai.model.builder, seems `create_model` requires the input model can only be huggingface model(remote or local with complete configs).
Could you help add a few comments to tell that if so? (please correct me if it is not only available for huggingface model.

Reply:
Yes, you are correct. Added a check in the pass to validate and raise an exception.